main.go

```diff
@@ -57,14 +60,47 @@ func init() {
 	utilruntime.Must(policyv1.AddToScheme(scheme))
 }
 
+type zapLevelFlag struct {
+	zapcore.Level
+}
+
+var _ pflag.Value = &zapLevelFlag{}
+
+// Set ensures that the level passed by a flag is valid
+func (f *zapLevelFlag) Set(val string) error {
```

Comment:
Using a custom flag type like this allows us to verify the input is valid when the flags are parsed. It might be worth having another one of these for zap-encoder, to ensure it is either `console` or `json`.

Reply:
Nice! I agree validating and throwing out invalid flags is a good idea rather than letting the controller crash.

Minor nit: Why not use `zapcore.Level.UnmarshalText()` (https://pkg.go.dev/go.uber.org/zap/zapcore#UnmarshalText)?

Reply:
Good suggestion!